### PR TITLE
Modernize test ASGI applications

### PR DESCRIPTION
**Pull request by florimondmanca**
Small refactoring PR that cleans up a bunch of test code, replacing the older ASGI 2.0 style:

```python
class App:
    def __init__(self, scope):
        ...

    async def __call__(self, receive, send):
        ...
```

With an up-to-date ASGI 3.0 style:

```python
async def app(scope, receive, send):
    ...
```

Also reduce code duplication in tests that were just using a sample "empty response" app.

### CODE REVIEW

File: tests/middleware/test_trace_logging.py [hunk 1]
```diff
@@ -48,26 +48,23 @@
...
-    class App:
-        def __init__(self, scope):
-            if scope["type"] != "http":
-                raise Exception()
-
-        async def __call__(self, receive, send):
-            await send({"type": "http.response.start", "status": 204, "headers": []})
-            await send({"type": "http.response.body", "body": b"", "more_body": False})
-
     class CustomServer(Server):
```
Comment (euri10) [inline tests/middleware/test_trace_logging.py:62]: I found myself having to change the CustomServer in 4 or 5 files while working on a draft today, maybe a good refactor for another pr to keep things separated or here have no preference, what do you think?

**Review by euri10 — APPROVED** [review]
Much love 💗 for net deletion, I left a small suggestion but this can go without !

File: tests/middleware/test_trace_logging.py (hunk 1 above)
Comment (florimondmanca) [reply tests/middleware/test_trace_logging.py:62]: Not sure what you mean here exactly — the `CustomServer` is left untouched? Only the ASGI apps are modified and factored out. Did you mean that the individual ASGI apps were modified as well?

**Comment by euri10** [comment]
Nah just wanted to say that in this PR or another the CustomServer could be
refactored as well, it's currently duplicated all over

**Comment by florimondmanca** [comment]
> in this PR or another the CustomServer could be refactored as well, it's currently duplicated all over

Yup ;-)
